packages/core/src/decorators/common/media-type-examples-override.ts

```diff
@@ -0,0 +1,41 @@
+import { Oas3Decorator } from '../../visitors';
+import { Oas3Operation, Oas3RequestBody } from '../../typings/openapi';
+import { readAndParseFileSync } from '../../utils';
+
+export const MediaTypeExamplesOverride: Oas3Decorator = ({ operationIds }) => {
+  let operationId: string | undefined;
+  return {
+    Operation: {
+      leave(operation: Oas3Operation) {
+        operationId = operation.operationId;
+        const properties = operationIds[operationId!];
+
+        if (properties.responses && operation.responses) {
+          for (const responseCode of Object.keys(properties.responses)) {
+            if (!operation.responses[responseCode] || !operation.responses[responseCode]?.content) {
+              continue;
+            }
+
+            Object.values(operation.responses[responseCode].content!).forEach((mimeType) => {
+              if (mimeType.examples) {
+                mimeType.examples = readAndParseFileSync(properties.responses[responseCode]);
```

Comment:
I'd suggest figuring out what type of example field is used, something like this:

```ts
const exampleField = mimeType.examples ? 'examples' : 'example';
mimeType[exampleField] = readAndParseFileSync(properties.responses[responseCode]);
```